embulk-jruby-extension/src/main/java/org/jruby/util/ParsedValues.java

```diff
@@ -0,0 +1,406 @@
+        ZONE_OFFSET_MAP.put("romance",                  3600);
+        ZONE_OFFSET_MAP.put("russian",                 10800);
+        ZONE_OFFSET_MAP.put("sa eastern",             -10800);
+        ZONE_OFFSET_MAP.put("sa pacific",             -18000);
+        ZONE_OFFSET_MAP.put("sa western",             -14400);
+        ZONE_OFFSET_MAP.put("samoa",                  -39600);
+        ZONE_OFFSET_MAP.put("se asia",                 25200);
+        ZONE_OFFSET_MAP.put("malay peninsula",         28800);
+        ZONE_OFFSET_MAP.put("south africa",             7200);
+        ZONE_OFFSET_MAP.put("sri lanka",               21600);
+        ZONE_OFFSET_MAP.put("taipei",                  28800);
+        ZONE_OFFSET_MAP.put("tasmania",                36000);
+        ZONE_OFFSET_MAP.put("tokyo",                   32400);
+        ZONE_OFFSET_MAP.put("tonga",                   46800);
+        ZONE_OFFSET_MAP.put("us eastern",             -18000);
+        ZONE_OFFSET_MAP.put("us mountain",            -25200);
+        ZONE_OFFSET_MAP.put("vladivostok",             36000);
+        ZONE_OFFSET_MAP.put("w. australia",            28800);
+        ZONE_OFFSET_MAP.put("w. central africa",        3600);
+        ZONE_OFFSET_MAP.put("w. europe",                3600);
+        ZONE_OFFSET_MAP.put("west asia",               18000);
+        ZONE_OFFSET_MAP.put("west pacific",            36000);
+        ZONE_OFFSET_MAP.put("yakutsk",                 32400);
+    }
+
+    // @see date_zone_to_diff in date_parse.c
+    // https://github.com/ruby/ruby/blob/trunk/ext/date/date_parse.c#L420
+    public static int toDiff(String zone)
+    {
+        String z = zone.toLowerCase();
+        int offset = Integer.MIN_VALUE;
+
+        boolean dst = false;
+        if (z.endsWith(" daylight time")) {
+            z = z.substring(0, z.length() - " daylight time".length());
+            dst = true;
+
+        } else if (z.endsWith(" standard time")) {
+            z = z.substring(0, z.length() - " standard time".length());
+            dst = false;
+
+        } else if (z.endsWith(" dst")) {
+            z = z.substring(0, z.length() - " dst".length());
+            dst = true;
+        }
+        if (ZONE_OFFSET_MAP.containsKey(z)) {
+            offset = ZONE_OFFSET_MAP.get(z);
+            if (dst) {
+                offset += 3600;
+            }
+            return offset;
+        }
+
+        if (z.startsWith("gmt") || z.startsWith("utc")) {
+            z = z.substring(3, z.length()); // remove "gmt" or "utc"
+        }
+
+        if (!isSign(z.charAt(0))) {
+            // if z doesn't have "+" or "-", invalid
+            return offset;
+        }
+
+        boolean sign = false;
+        sign = z.charAt(0) == '+';
+        z = z.substring(1, z.length());
+
+        int hour = 0, min = 0, sec = 0;
+        if (z.contains(":")) {
+            String[] splited = z.split(":");
+            if (splited.length == 2) {
+                hour = parseInt(splited[0]);
```

Comment:
oh, I was confused. This returns Integer.MIN_VALUE. This code is too much C style...